Customizing/global/plugins/Services/Cron/CronHook/UserOrguImport/classes/Filesystem/ImportFiles.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace CaT\IliasUserOrguImport\Filesystem;
+
+use CaT\IliasUserOrguImport\ErrorReporting\ErrorCollection as ErrorCollection;
+
+/**
+ * This class provides the means to import files to ilias from some
+ * deployment path and to look up the current soource-files.
+ */
+
+class ImportFiles
+{
+
+	const IMPORT_FILE_REGEX_CHECK = '#Check\\_[0-9]{4}\\-[0-9]{2}\\-[0-9]{2}\\.csv#';
+	const IMPORT_FILE_REGEX_DATA = '#SAP\\_[0-9]{4}\\-[0-9]{2}\\-[0-9]{2}\\.xlsx#';
+
+	protected $user;
+	protected $orgu;
+	protected $user_orgu;
+
+	protected $error_collection;
+	protected $filesystem;
+
+	public function __construct(Filesystem $fs, ErrorCollection $error_collection, FilesystemConfig $file_config)
+	{
+		$this->error_collection = $error_collection;
+		$this->filesystem = $fs;
+		$deployment_path = $file_config->deploymentPath();
+		if ($deployment_path) {
+			$this->importFiles($deployment_path, $this->filesystem);
+		}
+		$this->loadAndCheckCurrentFiles($this->filesystem);
```

Comment:
Please don't do stuff in the constructor besides real initialisation and checking of inputs. This is seen a lot in ILIAS, but IMO that is an antipattern. It makes reuse harder, potentially unnecessary work is done, and it's `__construct` after all, not `__constructAndPerformALotOfWork`.